src/pybind/mgr/cephadm/module.py

```diff
@@ -1490,6 +1490,21 @@ def get_hosts(self):
         """
         return list(self.inventory.all_specs())
 
+    @handle_orch_error
+    def get_gather_facts(self, hostname: Optional[str] = None) -> List[Dict[str, Any]]:
+        """
+        Return a list of hosts metadata(gather_facts) managed by the orchestrator.
+
+        Notes:
+          - skip async: manager reads from cache.
+        """
+        if hostname:
+            return [self.cache.get_facts(hostname)]
+        gather_facts = []
+        for h in self.cache.get_hosts():
+            gather_facts.append(self.cache.get_facts(h))
+        return gather_facts
```

Comment:
You could even simplify this to a one-liner (although it becomes less clear):
```suggestion
        return [self.cache.get_facts(hostname) for hostname in ([hostname] if hostname else self.cache.get_hosts())]
```